samples/CallWithChat/src/app/utils/useIsMobile.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { useEffect, useState } from 'react';
+import MobileDetect from 'mobile-detect';
+
+const detectMobileSession = (): boolean => !!new MobileDetect(window.navigator.userAgent).mobile();
+
+/**
+ * Detect if the app is running in a mobile browser and updated if that changes (e.g. when switching to a mobile browser in a browser emulator)
+ * @returns true if this detects the app is running in a mobile browser
+ */
+export const useIsMobile = (): boolean => {
```

Comment:
Very nice! Makes sense to have this as its own util.
Moving forward, it would still make sense to carry down the prop "isMobile" rather than utilizing this util in children elements right?

Reply:
We definitely shouldn't use this in the composites - so from that perspective it should be a passed down prop.
On the app side its up to you - there's pros and cons. I like "prop blasting" because then there is an interface where if you supply these props you get an exact UI. However prop blasting leads to hard-to-refactor code and overly verbose code... so up to you!